src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -385,6 +385,17 @@ def _validate_vm_create_storage_profile(cmd, namespace, for_scale_set=False):
     if not namespace.os_type:
         namespace.os_type = 'windows' if 'windows' in namespace.os_offer.lower() else 'linux'
 
+    if not namespace.os_caching:
+        # we will apply client side default till VMSS has the server side fault at ReadWrite.
+        from azure.cli.core.profiles import ResourceType
+        CachingTypes = cmd.get_models('CachingTypes', resource_type=ResourceType.MGMT_COMPUTE)
+        namespace.os_caching = CachingTypes.read_write.value
+
+    if namespace.data_caching and len(namespace.data_caching) > 1:
+        # verify if we have 2+ entries, all should have the format of '<lun>=<value>'
+        if len([x for x in namespace.data_caching if '=' in x]) != len(namespace.data_caching):
+            raise CLIError('usage error: --data-disk-caching VALUE | --data-disk-caching LUN=VALUE LUN2=VALUE2 ...')
```

Comment:
I'd suggest validating that the lefthand side is an int here and the right-hand side is a valid enum value.